xdem/volume.py

```diff
@@ -175,23 +233,45 @@ def calculate_hypsometry_area(ddem_bins: Union[pd.Series, pd.DataFrame], ref_dem
     return output
 
 
-def linear_interpolation(array: Union[np.ndarray, np.ma.masked_array]) -> np.ndarray:
+def linear_interpolation(array: Union[np.ndarray, np.ma.masked_array], max_search_distance: int = 10,
+                         extrapolate: bool = False, force_fill: bool = False) -> np.ndarray:
     """
-    Interpolate a 2D array using bilinear interpolation.
+    Interpolate a 2D array using rasterio's fillnodata.
 
     :param array: An array with NaNs or a masked array to interpolate.
+    :param max_search_distance: The maximum number of pixels to search in all directions to find values \
+to interpolate from. The default is 10.
+    :param extrapolate: if False, will remove pixels that have been extrapolated by fillnodata. Default is False.
+    :param force_fill: if True, will replace all remaining gaps by the median of all valid values. Default is False.
+
     :returns: A filled array with no NaNs
     """
     # Create a mask for where nans exist
-    nan_mask = (array.mask | np.isnan(array.data)) if isinstance(array, np.ma.masked_array) else np.isnan(array)
+    nan_mask = xdem.spatial_tools.get_mask(array)
+
+    interpolated_array = rasterio.fill.fillnodata(array.copy(), mask=(~nan_mask).astype("uint8"),
+                                                  max_search_distance=max_search_distance)
 
-    interpolated_array = rasterio.fill.fillnodata(array.copy(), mask=(~nan_mask).astype("uint8"))
+    # Remove extrapolated values
+    if not extrapolate:
+        interp_mask = cv2.morphologyEx((~nan_mask).squeeze().astype('uint8'), cv2.MORPH_CLOSE,
```

Comment:
I'd need a comment here to understand exactly what this does.
I'm not sure we need a closing, getting the dilation of `nan_mask` should be sufficient, no?

Reply:
It will remove all pixels that have information provided from one side only, i.e. extrapolated.
The closing makes sure that gaps the size of the search distance are kept, but all areas that have grown artificially are not.